### ISSUE TITLE
Mapping error on empty response from client.helpers().featureManagement().featureToggle

### ISSUE THREAD

**Issue by alansferreira** [hkarthik7/azure-devops-java-sdk#76]
Hi there, first then all, this library is awesome, but i'm receiving an exception when trying to disable some features of my projects like ARTIFACTS and BOARDS, maybe i'm doing something wrong.

Can you help me?

Steps to reproduce
------------------
my `pom.xml`

```xml

<dependency>
	<groupId>io.github.hkarthik7</groupId>
	<artifactId>azd</artifactId>
	<version>6.0.0</version>
</dependency>

```
```java
    var organizationUrl = Instance.BASE_INSTANCE.append("my-org");
    var pat = new PersonalAccessTokenCredential(organizationUrl, "my-project", "my-pat");
    
    var client = AzDService.builder().authentication(pat).buildClient();
    try {
      client.helpers().featureManagement().featureToggle(projectId, FeatureManagement.BOARDS, false);
    } catch (Exception e) {
      throw e;
    }

```

Expected behavior
-----------------

```none
just not error 
```

Actual behavior
---------------

```none
throws Exception 

ApiResponseParsingException: No content to map due to end-of-input
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1]
```

**Comment by hkarthik7** [comment]
Hiya @alansferreira, Thanks for your kind words and I'm sorry that you are facing this issue. The functionality is working, however, the response from Api when we run `client.helpers().featureManagement().featureToggle(projectId, FeatureManagement.BOARDS, false); ` is always `null` and I will push a fix for it.

Please run below and see if you are getting success response or not. In the below code `apiResponse.getStatusCode()` should return `OK`. Other two lines of code spit out the request url and the request body that was used to sent. The `ResponseHandler.getResponse()` is a container which holds useful information about the previously run request. This is useful for debugging.

```java
var projectId = client.core().projects().get("my-project").getId();
var projectFeature = client.helpers().featureManagement().featureToggle(projectId, FeatureManagement.BOARDS, true);

System.out.println(projectFeature); // Returns null and this is a bug. Will push a fix for this.

var apiResponse = ResponseHandler.getResponse(); 
var requestInformation = apiResponse.getRequestInformation();
String requestUrl = apiResponse.getRequestUrl();

System.out.println(apiResponse.getStatusCode());
System.out.println(requestUrl);
System.out.println(requestInformation.requestBody);
```

Please try using the old Api and see if you are getting the desired results.

```java
var webApi = new AzDClientApi("organisation", "project", "personalAccessToken");
var core = webApi.getCoreApi();

core.featureToggle("projectId", FeatureManagement.BOARDS, false);
```

**Comment by alansferreira** [comment]
Hi!! Im try to use old API, but without success, the same error was throws.

**Comment by hkarthik7** [comment]
Hi @alansferreira, Can you please run below and see what you are getting as response? The error message is that the response can't be deserialised or parsed properly. Can you please check the `personalAccessToken` that you are using has right permissions to the organisation/project and to toggle features?

```java
var projectId = client.core().projects().get("my-project").getId();
client.helpers().featureManagement().featureToggle(projectId, FeatureManagement.BOARDS, false);

System.out.println(ResponseHandler.getResponse().getResponseBody());
```

**Comment by alansferreira** [comment]
> Hi @alansferreira, Can you please run below and see what you are getting as response? The error message is that the response can't be deserialised or parsed properly. Can you please check the `personalAccessToken` that you are using has right permissions to the organisation/project and to toggle features?
> 
> ```java
> var projectId = client.core().projects().get("my-project").getId();
> client.helpers().featureManagement().featureToggle(projectId, FeatureManagement.BOARDS, false);
> 
> System.out.println(ResponseHandler.getResponse().getResponseBody());
> ```

Yes, my PAT has full permissions, manually configured one by one.

**Comment by hkarthik7** [comment]
Right! What is the response you are getting when you run that?

**Comment by hkarthik7** [comment]
Hi @alansferreira, Did you try to run the above and see what is the raw response that you got from API?

**Comment by hkarthik7** [comment]
Hi @alansferreira, Please use the latest version `6.0.1` and see what response you're getting from the API.

**Comment by alansferreira** [comment]
Thanks, i will try here!

**Comment by hkarthik7** [comment]
@alansferreira, Closing this issue as it was stale for few days now.
